### DISCUSSION TITLE
F2A FIDO since 1.18.0

### DISCUSSION THREAD

**Discussion by thelittlefireman** [dani-garcia/vaultwarden#1340] [Q&A]
reregister does nothing

### Subject of the issue
2FA FIDO not working in 1.18.0

### Your environment

* Bitwarden_rs version: 1.18.0

* Install method: Docker
* Clients used: web browser chrome
* Reverse proxy and version: nginx 1.18
* Version of mysql/postgresql: 
* Other relevant information:
DOMAIN YUBIKEY_API SERVER and secret redacted
Perfectly working in 1.17.0 i just upgrade the docker without change configuration

### Steps to reproduce
Activate 2FA in your account.
Disconect
Try login in web browser with 2FA FIDO
FIDO is loading infinitly

### Expected behaviour
2FA FIDO connection works

### Actual behaviour
FIDO is loading infinitly

### Relevant logs
Nothing special in logs, no message in web console like in #1300

![https://i.ibb.co/8Yh4fs5/screen.jpg](https://i.ibb.co/8Yh4fs5/screen.jpg)

**Comment by BlackDex** [comment]
I think error 5 means that either there's no https or the domain doesn't match. Please check those items first

**Reply by thelittlefireman** [reply]
hi thanks for the reply.
Still same error on 1.19

DOMAIN env is set https and certificate too. I don't change anything from 1.17.0 just upgrade the docker.

```
### Your environment (Generated via diagnostics page)
* Bitwarden_rs version: v1.19.0
* Web-vault version: v2.18.1
* Running within Docker: true
* Internet access: true
* Uses a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.*******.**",
  "domain_origin": "*****://*********.*******.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 9,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 30,
  "invitation_org_name": "Bitwarden_RS",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "log_file": "/data/bitwarden.log",
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "show_password_hint": true,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "Login",
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "********@*******.**",
  "smtp_from_name": "no_reply@mydomain.com",
  "smtp_host": "*******.**",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 30,
  "smtp_username": "********@*******.**",
  "templates_folder": "data/templates",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": "60272",
  "yubico_secret_key": "***",
  "yubico_server": "https://api2.yubico.com/wsapi/2.0/verify"
}
```

```

**Reply by BlackDex** [reply]
First of all, thanks for using the support string generator :).
That seems to look all good (Except the `smtp_from_name`, which should just be a name, and not an e-mail address).

Well, i tested this my self using a valid LetsEncrypt cert and a Self-Signed cert (localhost), and both work just great without any issues.
And i do see something in the developer console (F12).
![image](https://user-images.githubusercontent.com/610450/107155032-6961b300-6976-11eb-87ed-abd868c03786.png)

That is something you should see also, if not, then it looks like there is something blocking from the browser, or the server.
Could you double check what you see in the Developer Console and drop a screenshot or copy the text here?

Thanks

**Reply by thelittlefireman** [reply]
Thanks for your reply :)

that's all i've got on dev console with a full login :/


EDIT : OH damned i found my mistake ! Sorry it was an nginx miss-config (forgot to change `x-frame-options` from `deny` to `sameorigin` )

thanks for the help :)
